src/components/TodosContext.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from 'react';
+
+import { Todo } from '../types/Todo';
+import { Context } from '../types/Context';
+import { Filter } from '../types/Filter';
+
+export const TodosContext = React.createContext<Context>({
+  todos: [],
+  addTodo: () => { },
+  deleteTodo: () => { },
+  editingTodo: () => { },
+  toggleTodo: () => { },
+  deleteCompletedTodos: () => { },
+  handleToggleAll: () => { },
+  todoCount: 0,
+  completedTodos: false,
+  filterTodos: () => [],
+});
```

Comment:
save in the Context only data that is used in several components on different levels(with redundant lifting). For example, you can calculate `todoCount` directly in the App based on `todos`. Storing in Component is cheaper than storing in Context, so don't overuse it